pkg/ccl/changefeedccl/cdctest/validator_helper.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package cdctest
+
+import (
+	gojson "encoding/json"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/json"
+	"github.com/cockroachdb/cockroach/pkg/workload/rand"
+)
+
+func extractFieldFromJSON(d json.JSON) (interface{}, error) {
+	switch d.Type() {
+	case json.ArrayJSONType:
+		iter, _ := d.AsArray()
+		ret := make([]interface{}, 0)
+		for i := 0; i < len(iter); i++ {
+			if i, err := extractFieldFromJSON(iter[i]); err == nil {
+				ret = append(ret, i)
+			}
+		}
+		return ret, nil
+	case json.ObjectJSONType:
+		if g, err := geo.ParseGeographyFromGeoJSON([]byte(d.String())); err == nil {
+			return rand.DatumToGoSQL(tree.NewDGeography(g))
+		}
+		if g, err := geo.ParseGeometryFromGeoJSON([]byte(d.String())); err == nil {
+			return rand.DatumToGoSQL(tree.NewDGeometry(g))
+		}
+	default:
+		break
+	}
+	var ret interface{}
+	err := gojson.Unmarshal([]byte(d.String()), &ret)
+	return ret, err
+}
+
+func parseJSON(d json.JSON) (map[string]interface{}, error) {
```

Comment:
i don't love this function's name -- it takes an already parsed crdb json and transforms it into go datatypes, including some weird geometry parsing stuff. can you update the name to reflect this?

Reply:
Renamed it to convertJSONToMap. Do you like this more?